File: flutter_app/lib/services/api_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

class ApiService {

  static const String baseUrl = "http://10.0.2.2:8000"; // Android emulator

  static Future<List<dynamic>> fetchDailyCA() async {
    final response = await http.get(
      Uri.parse("$baseUrl/ca/daily"),
    );

    if (response.statusCode == 200) {
      final data = jsonDecode(response.body);
      return data["data"];
    } else {
      throw Exception("Failed to load current affairs");
    }
  }
}

File: frontend_streamlit/services/api_service.py
# =========================================================
# 🌐 API SERVICE (STREAMLIT → FASTAPI CONNECTOR)
# =========================================================

import requests


# =========================================================
# ⚙️ CONFIG
# =========================================================
BASE_URL = "http://127.0.0.1:8000"


# =========================================================
# 📰 GET CURRENT AFFAIRS
# =========================================================
def get_daily_current_affairs(token: str = None):

    try:

        headers = {}

        if token:
            headers["Authorization"] = f"Bearer {token}"

        response = requests.get(
            f"{BASE_URL}/ca/daily",
            headers=headers,
            timeout=10
        )

        if response.status_code != 200:
            print("[API ERROR]", response.text)
            return []

        data = response.json()

        return data.get("data", [])

    except Exception as e:
        print("[API FAILED]", e)
        return []